blocks/library/shortcode/index.js

```diff
@@ -31,6 +31,27 @@ registerBlockType( 'core/shortcode', {
 		},
 	},
 
+	transforms: {
+		from: [
+			{
+				type: 'shortcode',
+				// Per "Shortcode names should be all lowercase and use all
+				// letters, but numbers and underscores should work fine too.
+				// Be wary of using hyphens (dashes), you'll be better off not
+				// using them." in https://codex.wordpress.org/Shortcode_API
+				tag: '[A-z0-9_-]+',
```

Comment:
Per above comment, should we not be matching capital letters as the regular expression here currently allows? Wondering if there's a more canonical regular expression from core code we should be leveraging, or if we should bother being strict at all.

Reply:
Yes, I'm fine with keeping it lowercase.